tests/ExpressionParserTest.php

```diff
@@ -208,6 +209,142 @@ public function getTestsForSequence()
         ];
     }
 
+    /** @dataProvider getTestsForNameToStringMapping */
+    public function testNameToStringMapping(string $template, ArrayExpression $expected, bool $allowOptionalKeys): void
+    {
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $parser = new Parser($env);
+        $stream = $env->tokenize(new Source($template, ''));
+        $stream->next();
+
+        $expressionParser = new ExpressionParser($parser, $env);
+
+        $this->assertEquals($expected, $expressionParser->parseNameToStringMapping($stream, $allowOptionalKeys));
+    }
+
+    public function getTestsForNameToStringMapping(): array
+    {
+        return [
+            // empty map
+            [
+                '{{ {} }}',
```

Comment:
Those tests don't make sense as they don't parse the source that is visible here (btw, those hacky tests seem to be the primary reason why the design of `ExpressionParser::parseNameToStringMapping` does not match the design of other parts of the Twig parser).

Your tests should be parsing a `{% types %}` tag and asserting its structure instead.

Reply:
You're spot on re: test design.

> Your tests should be parsing a {% types %} tag and asserting its structure instead.

Yep, will do!